assets/scripts/palette/Palette.jsx

```diff
@@ -28,63 +23,54 @@ function Palette (props) {
     }
   }, [])
 
-  function renderPaletteItems () {
-    const segments = getAllSegmentInfoArray()
-
-    // For each segment, set "disabled" property instead that indicates
-    // whether this segment is in a disabled state for this user
-    // Then filter out disabled segments that do not have the
-    // `alwaysShowInPalette` property set to `true`
-    const displayedSegments = segments
-      .map((segment) => {
-        // Accept segments that don't have the `enableWithFlag` property
-        const enabledByDefault = !segment.enableWithFlag
-        // Accept segments with the `enableWithFlag` property, but only if
-        // the flags have that value set to true.
-        const enabledByFlag =
-          segment.enableWithFlag && flags[segment.enableWithFlag].value
+  const segments = getAllSegmentInfoArray()
 
-        return {
-          ...segment,
-          disabled: !(enabledByDefault || enabledByFlag)
-        }
-      })
-      .filter(
-        (segment) =>
-          !segment.disabled || (segment.disabled && segment.alwaysShowInPalette)
-      )
-
-    // Return all enabled segments as an array of SegmentForPalette components.
-    return displayedSegments.map((segment) => {
-      const variant = segment.paletteIcon
-        ? segment.paletteIcon
-        : Object.keys(segment.details).shift()
+  // For each segment, set "disabled" property instead that indicates
+  // whether this segment is in a disabled state for this user
+  // Then filter out disabled segments that do not have the
+  // `alwaysShowInPalette` property set to `true`
+  const displayedSegments = segments
+    .map((segment) => {
+      // Accept segments that don't have the `enableWithFlag` property
+      const enabledByDefault = !segment.enableWithFlag
+      // Accept segments with the `enableWithFlag` property, but only if
+      // the flags have that value set to true.
+      const enabledByFlag =
+        segment.enableWithFlag && flags[segment.enableWithFlag].value
 
-      return (
-        <SegmentForPalette
-          key={segment.id}
-          type={segment.id}
-          variantString={variant}
-          onPointerOver={props.handlePointerOver}
-          randSeed={randSeed.current}
-          disabled={segment.disabled}
-        />
-      )
+      return {
+        ...segment,
+        disabled: !(enabledByDefault || enabledByFlag)
+      }
     })
-  }
+    .filter(
+      (segment) =>
+        !segment.disabled || (segment.disabled && segment.alwaysShowInPalette)
+    )
+    .map((segment) => (
+      <SegmentForPalette
```

Comment:
This is unrelated to the PR, but I'm noticing there are about twice as many of these components rendered as I'd expect. Something to do with the drag-able versions of the component?

Reply:
Say more? What do you mean by twice?

Are there twice as many items in the UI? Or is it rendering twice?

Reply:
When inspecting components, I see around 48 SegementForPalette which is more than we have available, no?

Reply:
@easherma That shouldn't be happening. Do you see 48 segments in the palette (visually, not by inspecting in devtools)? Is this a bug that's introduced by this PR?

Reply:
Oh - I see you mentioned it's not related to the PR. Let's take this conversation to Discord?